Typing full `/delta` command and pressing `space` key crash the ide

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Typing full `/delta` command and pressing space key crash the editor. Typing `/delt` and pressing tab do nothing.

### Environment

Zed: v0.158.1 (Zed)
OS: macOS 14.6.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-10-24T17:47:55.608804+02:00 [INFO] add connection to peer
2024-10-24T17:47:55.609075+02:00 [INFO] add_connection;
2024-10-24T17:47:55.609291+02:00 [INFO] waiting for server hello
2024-10-24T17:47:55.609696+02:00 [INFO] got server hello
2024-10-24T17:47:55.609883+02:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 601, id: 50944 })
2024-10-24T17:47:55.610211+02:00 [INFO] set status on client 178985: Connected { peer_id: PeerId { owner_id: 601, id: 50944 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-10-24T17:47:55.634262+02:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-10-24T15:47:55.629Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-10-24T15:47:55.629Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-10-24T18:09:00.587369+02:00 [ERROR] {
  "thread": "main",
  "payload": "not implemented",
  "location_data": {
    "file": "crates/assistant/src/slash_command/delta_command.rs",
    "line": 40
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "zed::reliability::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::panicking::panic",
    "<assistant::slash_command::delta_command::DeltaSlashCommand as assistant_slash_command::SlashCommand>::complete_argument",
    "<assistant::slash_command::SlashCommandCompletionProvider as editor::CompletionProvider>::completions",
    "<editor::Editor>::show_completions",
    "<editor::Editor>::handle_input::{closure#8}",
    "<editor::Editor>::handle_input",
    "<editor::Editor as gpui::input::ViewInputHandler>::replace_text_in_range",
    "<gpui::input::ElementInputHandler<editor::Editor> as gpui::platform::InputHandler>::replace_text_in_range",
    "<gpui::platform::PlatformInputHandler>::replace_text_in_range",
    "gpui::platform::mac::window::send_to_input_handler",
    "gpui::platform::mac::window::handle_key_event",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run",
    "<gpui::app::App>::run::<zed::main::{closure#5}>",
    "zed::main",
    "std::sys::backtrace::__rust_begin_short_backtrace::<fn(), ()>",
    "std::rt::lang_start::<()>::{closure#0}",
    "std::rt::lang_start_internal",
    "_main"
  ],
  "app_version": "0.158.1",
  "release_channel": "Zed",
  "os_name": "macOS",
  "os_version": "14.6.1",
  "architecture": "aarch64",
  "panicked_on": 1729786140587,
  "system_id": "9d9632dc-65ea-40f7-851e-491bb44c4435",
  "installation_id": "28545d90-072c-480a-b3b3-da542b476942",
  "session_id": "67e24136-ecba-4582-b4e8-0733cac5a5d4"
}
2024-10-24T18:09:03.18178+02:00 [INFO] ========== starting zed ==========
2024-10-24T18:09:03.182897+02:00 [INFO] Opening main db
2024-10-24T18:09:03.184499+02:00 [INFO] Opening main db
2024-10-24T18:09:03.2106+02:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2024-10-24T18:09:03.536732+02:00 [INFO] no tokio runtime found, creating one for Reqwest...
2024-10-24T18:09:03.56142+02:00 [ERROR] theme not found: GitHub Dark Default
2024-10-24T18:09:03.56185+02:00 [INFO] initializing context server client
2024-10-24T18:09:03.562016+02:00 [INFO] Registering built-in prompt template: content_prompt
2024-10-24T18:09:03.562434+02:00 [INFO] Registering built-in prompt template: edit_workflow
2024-10-24T18:09:03.562642+02:00 [INFO] Registering built-in prompt template: project_slash_command
2024-10-24T18:09:03.562722+02:00 [INFO] Registering built-in prompt template: step_resolution
2024-10-24T18:09:03.56308+02:00 [INFO] Registering built-in prompt template: terminal_assistant_prompt
2024-10-24T18:09:03.563453+02:00 [INFO] No prompt template overrides directory found at /Users/rafalmoneta/.config/zed/prompt_overrides. Using built-in prompts.
2024-10-24T18:09:03.563974+02:00 [INFO] extensions updated. loading 10, reloading 0, unloading 0
2024-10-24T18:09:03.612464+02:00 [INFO] Opening main db
2024-10-24T18:09:03.619636+02:00 [WARN] Theme "Dracula" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-10-24T18:09:03.62616+02:00 [WARN] Theme "GitHub Dark Default" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-10-24T18:09:03.639347+02:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-10-24T18:09:03.701171+02:00 [WARN] Theme "The Dark Side" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-10-24T18:09:03.808304+02:00 [INFO] set status on client 0: Authenticating
2024-10-24T18:09:03.866583+02:00 [INFO] set status on client 178985: Connecting
2024-10-24T18:09:03.886473+02:00 [INFO] Opening main db
2024-10-24T18:09:04.211368+02:00 [INFO] Node runtime install_if_needed
2024-10-24T18:09:04.383002+02:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-10-24T18:09:04.422895+02:00 [INFO] starting language server process. binary path: "/Users/rafalmoneta/Library/Application Support/Zed/node/node-v22.5.1-darwin-arm64/bin/node", working directory: "/", args: ["/Users/rafalmoneta/Library/Application Support/Zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-10-24T18:09:04.6729+02:00 [INFO] set environment variables from shell:/bin/zsh, path:/opt/homebrew/opt/sqlite/bin:/Users/rafalmoneta/.amplify/bin:/bin:/usr/bin:/usr/local/bin:/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/rafalmoneta/.nvm/versions/node/v20.16.0/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/usr/local/go/bin:/Users/rafalmoneta/.cargo/bin:/Applications/Visual Studio Code.app/Contents/Resources/app/bin:/Users/rafalmoneta/go/bin
2024-10-24T18:09:04.846138+02:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-10-24T16:09:04.841Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-10-24T16:09:04.841Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-10-24T18:09:04.853686+02:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "ab08030f-e2c3-4461-8b8a-623c1a8077ac",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-10-24T18:09:05.30248+02:00 [INFO] using project environment variables shell launched in "/Users/rafalmoneta/Code/rafmon.dev". PATH="/opt/homebrew/opt/sqlite/bin:/Users/rafalmoneta/.amplify/bin:/bin:/usr/bin:/usr/local/bin:/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/rafalmoneta/.nvm/versions/node/v20.16.0/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/usr/local/go/bin:/Users/rafalmoneta/.cargo/bin:/Applications/Visual Studio Code.app/Contents/Resources/app/bin:/Users/rafalmoneta/go/bin"
2024-10-24T18:09:05.410556+02:00 [INFO] add connection to peer
2024-10-24T18:09:05.410647+02:00 [INFO] add_connection;
2024-10-24T18:09:05.410728+02:00 [INFO] waiting for server hello
2024-10-24T18:09:05.419089+02:00 [INFO] got server hello
2024-10-24T18:09:05.419246+02:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 601, id: 64018 })
2024-10-24T18:09:05.419363+02:00 [INFO] set status on client 178985: Connected { peer_id: PeerId { owner_id: 601, id: 64018 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/assistant/src/slash_command/delta_command.rs
+++ b/crates/assistant/src/slash_command/delta_command.rs
@@ -1,5 +1,5 @@
 use crate::slash_command::file_command::{FileCommandMetadata, FileSlashCommand};
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use assistant_slash_command::{
     ArgumentCompletion, SlashCommand, SlashCommandOutput, SlashCommandOutputSection,
     SlashCommandResult,
@@ -38,7 +38,7 @@ impl SlashCommand for DeltaSlashCommand {
         _workspace: Option<WeakView<Workspace>>,
         _cx: &mut WindowContext,
     ) -> Task<Result<Vec<ArgumentCompletion>>> {
-        unimplemented!()
+        Task::ready(Err(anyhow!("this command does not require argument")))
     }
 
     fn run(

